cmd/telegram-butler/setup_logger.go

```diff
@@ -0,0 +1,22 @@
+package main
+
+import (
+	"log/slog"
+	"os"
+
+	"github.com/spf13/viper"
+
+	"github.com/GolangUA/telegram-butler/internal/module/logger"
+)
+
+func setupLogger() *slog.Logger {
```

Comment:
@vtopc 

I decided to use the `setup` instead of `new` because the name `setupLogger` implies that the function not only creates a new logger instance but also configures it with predefined settings (like log level, output format, etc.)

This approach enhances code clarity and uniformity, ensuring the `logger` is ready for use with all necessary configurations immediately after initialization

Reply:
> but also configures it with predefined settings (like log level, output format, etc.)

boilerplate?

Reply:
> > but also configures it with predefined settings (like log level, output format, etc.)
> 
> boilerplate?

The term '_boilerplate_' often refers to sections of code that are repeated in multiple places with little to no variation.

However, in the case of `setupLogger`, it is foundational in ensuring that all logger instances across our application are consistent and configured correctly.
By centralizing our logger configuration, we reduce the risk of discrepancies and make it easier to adjust settings globally. 

Rather than boilerplate, I see it as an essential part of our logging infrastructure that reinforces best practices and maintainability.

Reply:
Folks, I'm not sure if this code shall be in this pkg at all.
What do we have here? - Log initialisation with some options. I don't see any relations to the thing we have in our main/setup files.
This function, if we make it public, could be used from many different pkgs with no issues. 

Is there any reason why we shouldn't move this to a separate log pkg?

Reply:
In general, we have to keep in mind that we want to achieve the high level of decomposition. 
If there is no reasons for keeping things together - we have to move them to separate pkgs and make them to be isolated.

Reply:
@vpakhuchyi, yeah, looks like it should be in `github.com/GolangUA/telegram-butler/internal/module/logger`

Reply:
@vtopc oki doki)
You got it